### ISSUE TITLE
Support for idp with self signed certificates

### ISSUE THREAD

**Issue by varunsharma27** [hapijs/bell#444]
#### Support plan

* *which support plan is this issue covered by?* (e.g. Community, Core, Plus, or Enterprise): Community
* *is this issue currently blocking your project?* (yes/no): yes
* *is this issue affecting a production system?* (yes/no): yes

#### Context

* *node version*: v10.14.2
* *module version*: ^8.9.0
* *environment* (e.g. node, browser, native): Node
* *used with* (e.g. hapi application, another framework, standalone, ...): HapiJS v14.2.0

#### What problem are you trying to solve?

Currently the discovery or requests to the idp will fail it is running with a self-signed certificate or a certificate from an unknown CA. To fix such a situation, we'd require ability to provide custom certificates to use instead of the system CAs and ability to have different Verification levels.
```js
	none:
		rejectUnauthorized: false;
	full: //hostname + certificate verification
		rejectUnauthorized: true;
	certificate: //certificate only verification
		rejectUnauthorized: true;
		tls.checkServerIdentity: _.noop;
		
```

#### Do you have a new or modified API suggestion to solve the problem?
As far as I can see, the choices basically would be:

- Add an option to Bell's configuration to specify a custom Wreck instance (by default set to the singleton)
- Add an option to Bell's configuration with brutal requestOptions overrides
- Add TLS verification specific options and modify Bell to set requestOptions accordingly

The easiest solution seems to be adding the ability to provide a custom Wreck instance, here the users can pretty much do anything they like with most flexibility. The other two options would make Bell easier to use for others with the last option being the most comprehensive.

If we can come up with a consensus then I'd be surely willing to open up a PR implementing the discussed solution along with relevant test cases.

**Comment by hueniverse** [comment]
You are aware that hapi v14 has not been supported for a long time...? Any feature added to bell will not really solve your issue unless you also upgrade to hapi v18.

**Comment by varunsharma27** [comment]
Yes, a HapiJS upgrade is in pipeline. And I can work with a forked branch till we upgrade but simultaneously make PR against bell master so the future is clear.
